src/services/subscription/subscription-router.test.ts

```diff
@@ -1,113 +1,130 @@
 import { IncomingSubscription } from "./subscription-schema";
-import { SendEmailCommand } from "@aws-sdk/client-sesv2";
 import Config from "../../config";
 
-const EMAIL_1 = "testuser@example.com";
-const EMAIL_2 = "otheruser@example.com";
-const INVALID_EMAIL = "not-an-email";
+// Mock the AWS SDK
+const mockSESV2Send = jest.fn();
+const mockSendEmailCommand = jest.fn((input) => input);
+jest.mock("@aws-sdk/client-sesv2", () => ({
+    SESv2Client: jest.fn(() => ({
+        send: mockSESV2Send,
+    })),
+    SendEmailCommand: mockSendEmailCommand,
+}));
+
+const USER_ID_1 = "test-er-user-id";
+const USER_ID_2 = "test-user-2";
+const INVALID_USER_ID = "invalid-user-id";
 const VALID_mailingList = "rp_interest";
 const INVALID_mailingList = "invalid_list";
 
 const SUBSCRIPTION_1 = {
-    email: EMAIL_1,
+    userId: USER_ID_1,
     mailingList: VALID_mailingList,
 } satisfies IncomingSubscription;
 const SUBSCRIPTION_2 = {
-    email: EMAIL_2,
+    userId: USER_ID_2,
     mailingList: VALID_mailingList,
 } satisfies IncomingSubscription;
-const SUBSCRIPTION_INVALID_EMAIL = {
-    email: INVALID_EMAIL,
+const SUBSCRIPTION_INVALID_USER_ID = {
+    userId: INVALID_USER_ID,
     mailingList: VALID_mailingList,
 };
 const SUBSCRIPTION_INVALID_LIST = {
-    email: EMAIL_1,
+    userId: USER_ID_1,
     mailingList: INVALID_mailingList,
 };
 
-const mockSESV2Send = jest.fn();
-jest.mock("@aws-sdk/client-sesv2", () => {
-    return {
-        SESv2Client: jest.fn(() => ({
-            send: mockSESV2Send,
-        })),
-        SendEmailCommand: jest.fn((input) => input), // Mocks the command to return its input
-    };
-});
-
 beforeEach(async () => {
     jest.clearAllMocks();
+
+    // Set up authInfo data for test users
+    try {
+        await SupabaseDB.AUTH_INFO.insert([
+            {
+                userId: USER_ID_1,
+                email: "user1@test.com",
+                authId: "auth1",
+                displayName: "User 1",
+            },
+            {
+                userId: USER_ID_2,
+                email: "user2@test.com",
+                authId: "auth2",
+                displayName: "User 2",
+            },
+        ]);
+    } catch {
+        // Ignore errors if data already exists
+    }
 });
 
 describe("POST /subscription/", () => {
-    it("should create a new subscription for a new mailing list", async () => {
+    it("should create a new subscription for a new user", async () => {
         const response = await post("/subscription/")
             .send(SUBSCRIPTION_1)
             .expect(StatusCodes.CREATED);
         expect(response.body).toEqual(SUBSCRIPTION_1);
-        const { data } = await SupabaseDB.SUBSCRIPTIONS.select().eq(
-            "mailingList",
-            VALID_mailingList
-        );
+        const { data } = await SupabaseDB.SUBSCRIPTIONS.select()
+            .eq("userId", USER_ID_1)
+            .eq("mailingList", VALID_mailingList);
         const dbEntry = data?.[0];
```

Comment:
Use `.maybeSingle` for any calls where you expect one or none response, and use `.throwOnError` for all db calls so we get good logging.